ui/store/actions.js

```diff
@@ -624,196 +625,36 @@ export function signTypedMsg(msgData) {
 }
 
 export function signTx(txData) {
-  return (dispatch) => {
+  return async (dispatch) => {
+    dispatch(showLoadingIndication());
     global.ethQuery.sendTransaction(txData, (err) => {
       if (err) {
         dispatch(displayWarning(err.message));
       }
     });
+    await wait(500);
```

Comment:
This wait is necessary because there is no guarantee that after the token.transfer call is made that the transaction has fully made its way into the unapproved transaction state. I am not sure why moving the send state out of the metamask state makes this issue happen but the way it sufficient to rectify it. Tried with 200,300 and 400.500 is successful 100% of the time while 400 had intermittent test failures.

Reply:
Possibly related, though I'm not sure how yet, but.... it looks like the call to `showConfTxPage` does nothing but hide the loading indicator and clear any warnings that are set (if any). It doesn't do anything. It does not show the confirm transaction page :confused: 

So that has nothing to do with the change in behaviour here. It must be in the UI component.

Reply:
in send-footer.component, onSubmit:
```js
    Promise.resolve(promise).then(() => {
      metricsEvent({
        eventOpts: {
          category: 'Transactions',
          action: 'Edit Screen',
          name: 'Complete',
        },
        customVariables: {
          gasChanged: gasEstimateType,
        },
      });
      history.push(CONFIRM_TRANSACTION_ROUTE);
    });
```

Reply:
what's weird to me is before I started messing with this, the 'sign' method didn't return a promise... just the 'update' method. At least that's how I interpreted it.

Reply:
oh haha, Promise.resolve(promise) is just going to return a resolved promise that resolves whatever is passed in 🤦🏻